Tasks/DotNetCoreInstallerV0/dotnetcoreinstaller.ts

```diff
@@ -3,28 +3,80 @@ import * as toolLib from 'vsts-task-tool-lib/tool';
 import * as trm from 'vsts-task-lib/toolrunner';
 import { DotNetCoreReleaseFetcher } from "./releasesfetcher";
 import * as utilities from "./utilities";
-
+import * as fileSystem from "fs";
 import * as os from 'os';
 import * as path from 'path';
 
 class DotnetCoreInstaller {
-    constructor(packageType, version) {
+    constructor(workingDirectory: string, packageType: string, version?: string, useGlobalJson?: boolean) {
+        this.useGlobalJson = useGlobalJson;
         this.packageType = packageType;
+        this.workingDirectory = workingDirectory;
         if (!toolLib.isExplicitVersion(version)) {
             throw tl.loc("ImplicitVersionNotSupported", version);
         }
+        if(!useGlobalJson && (version == null || version.length == 0)){
+            throw tl.loc("VersionMissingException", "If you don't use global.json then please defined a version.")
```

Comment:
tl.loc("VersionMissingException") this would work, you need not rewrite the message "if you don't ..." again here.

Reply:
done.

Reply:
I think we have resolve this. If not reopen this conversion.